cmake/CMakeLists.txt

```diff
@@ -575,118 +594,128 @@ if(NOT MINIMAL_BUILD AND HDF5_SUPPORT)
                 set(lib_ext so)
             endif()
 
-            SET(HDF5_LIBRARIES
-                    "${HDF5_LIBRARY_DIR}/libhdf5_cpp.${lib_ext}"
-                    "${HDF5_LIBRARY_DIR}/libhdf5.${lib_ext}"
-                    "${HDF5_LIBRARY_DIR}/${zlib_post}.${lib_ext}"
-                    )
+            set(HDF5_LIBRARIES
+                "${HDF5_LIBRARY_DIR}/libhdf5_cpp.${lib_ext}"
+                "${HDF5_LIBRARY_DIR}/libhdf5.${lib_ext}"
+                "${HDF5_LIBRARY_DIR}/${zlib_post}.${lib_ext}"
+            )
         endif()
-
     endif()
 
-    if (NOT HDF5_LIBRARIES)
-    	# debian default
-
-        SET(HDF5_LIBRARIES
+    if(NOT HDF5_LIBRARIES)
+        # debian default
+        set(HDF5_LIBRARIES
             /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5_cpp.so
             /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so
             /usr/lib/x86_64-linux-gnu/libsz.so
             /usr/lib/x86_64-linux-gnu/libaec.so
             z dl
         )
     endif()
-    
+
     add_definitions(-DWITH_HDF5)
-	set(SWIG_DEFINES ${SWIG_DEFINES} -DWITH_HDF5)
+    set(SWIG_DEFINES ${SWIG_DEFINES} -DWITH_HDF5)
 endif()
 
```

Comment:
Preparation of external library paths is done here. Add reset of "CMAKE_FIND_ROOT_PATH" here back in, right before other actions like complier settings etc.